code/game/machinery/doors/door.dm

```diff
@@ -76,8 +76,14 @@
 		var/mob/living/M = AM
 		if(world.time - M.last_bumped <= 10) return	//Can bump-open one airlock per second. This is to prevent shock spam.
 		M.last_bumped = world.time
-		if(!M.restrained() && M.mob_size > MOB_SIZE_SMALL)
-			bumpopen(M)
+		if(!M.restrained())
+			if(ispet(AM))
+				var/mob/living/simple_animal/A = AM
+				if(A.collar)
+					bumpopen(M)
+			else

```

Comment:
`else if ()` is better than
```
else
    if()
```

Reply:
seconded, this should be one line

Reply:
Shouldn't the bumpopens actually use AM?